packages/grant-explorer/src/features/projects/hooks/useProject.ts

```diff
@@ -0,0 +1,18 @@
+import useSWR from "swr";
+import { DataLayer } from "data-layer";
+import { AlloVersion } from "data-layer/src/data-layer.types";
+
+type Params = {
+  projectId?: string;
+};
+
+export function useProject(params: Params, dataLayer: DataLayer) {
+  const shouldFetch = Object.values(params).every(Boolean);
+  return useSWR(shouldFetch ? ["applications", params] : null, async () => {
+    const validatedParams = {
+      projectId: params.projectId as string,
```

Comment:
maybe here we can use `projectId!` since we already checked that it's not undefined